### PR TITLE
feat: static/dynamic-external schema handling

### PR DESCRIPTION
**Pull request by zeljkobekcic**
This branch implements a possibility to load Precise's schema from an external file if a property is set.

### CODE REVIEW

File: sdk-autoconfigure/src/main/kotlin/io/datalbry/connector/autoconfigure/bootstrap/AlxndriaSchemaBootstrapAutoConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import io.datalbry.connector.sdk.ConnectorProperties
+import io.datalbry.connector.sdk.bootstrap.AlxndriaSchemaBootstrap
+import io.datalbry.connector.sdk.schema.SchemaProvider
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+@ConditionalOnClass(PlatformClient::class, SchemaProvider::class, ConnectorProperties::class)
+open class AlxndriaSchemaBootstrapAutoConfiguration {
```
Comment (Lacritz) [inline sdk-autoconfigure/src/main/kotlin/io/datalbry/connector/autoconfigure/bootstrap/AlxndriaSchemaBootstrapAutoConfiguration.kt:13]: Nope. Has nothing to do with Alxndria. It's a Precise Schema not a Alxndria schema

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/bootstrap/AlxndriaSchemaBootstrap.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * The [AlxndriaSchemaBootstrap] bootstraps alxndria with a schema. Upon start up the  loaded schema will be put into
+ * axlndria.
+ *
+ * @param alxndria: The alxndria client
+ * @param schemaProvider: A schema provider.
+ * @param properties: Connector properties which contain information about alxndria's datasource.
+ */
+open class AlxndriaSchemaBootstrap(
```
Comment (Lacritz) [inline sdk/src/main/kotlin/io/datalbry/connector/sdk/bootstrap/AlxndriaSchemaBootstrap.kt:17]: s.a.

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ItemMapper.kt [hunk 3]
```diff
@@ -3,7 +3,7 @@ package io.datalbry.connector.sdk.consumer.generic
 import io.datalbry.precise.api.schema.document.Document
 import io.datalbry.connector.api.DocumentEdge
 
-internal interface ItemMapper<T> {
+interface ItemMapper<T> {
```
Comment (Lacritz) [inline sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ItemMapper.kt:6]: missing kdoc

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ReflectionProcessor.kt [hunk 4]
```diff
@@ -2,7 +2,7 @@ package io.datalbry.connector.sdk.consumer.generic
 
 import io.datalbry.connector.api.document.DocumentProcessor
 
-internal interface ReflectionProcessor: DocumentProcessor<Any, Any> {
+interface ReflectionProcessor: DocumentProcessor<Any, Any> {
```
Comment (Lacritz) [inline sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ReflectionProcessor.kt:5]: missing kdoc

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/schema/external/ExternalConfiguredSchemaProviderProperties.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    @PropertyDescription(
+        key = "io.datalbry.schema.external.path",
+        group = "External Precise Schema",
+        type = STRING_KEY,
+        description = "The URL for the sample API instance",
+        label = "Filepath",
+        required = true
+    )
+    val path: String
+)
```
Comment (Lacritz) [inline sdk/src/main/kotlin/io/datalbry/connector/sdk/schema/external/ExternalConfiguredSchemaProviderProperties.kt:22]: missing line feed at the end of the file

File: sdk-autoconfigure/src/main/kotlin/io/datalbry/connector/autoconfigure/bootstrap/AlxndriaSchemaBootstrapAutoConfiguration.kt (hunk 1 above)
Comment (zeljkobekcic) [reply sdk-autoconfigure/src/main/kotlin/io/datalbry/connector/autoconfigure/bootstrap/AlxndriaSchemaBootstrapAutoConfiguration.kt]: I have renamed it accordingly. Thank you for you feedback!

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/bootstrap/AlxndriaSchemaBootstrap.kt (hunk 2 above)
Comment (zeljkobekcic) [reply sdk/src/main/kotlin/io/datalbry/connector/sdk/bootstrap/AlxndriaSchemaBootstrap.kt]: s.a.

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ReflectionProcessor.kt (hunk 4 above)
Comment (zeljkobekcic) [reply sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ReflectionProcessor.kt:8]: Done

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/schema/external/ExternalConfiguredSchemaProviderProperties.kt (hunk 5 above)
Comment (zeljkobekcic) [reply sdk/src/main/kotlin/io/datalbry/connector/sdk/schema/external/ExternalConfiguredSchemaProviderProperties.kt:22]: Done

File: sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ItemMapper.kt (hunk 3 above)
Comment (zeljkobekcic) [reply sdk/src/main/kotlin/io/datalbry/connector/sdk/consumer/generic/ItemMapper.kt:9]: done

**Pull request merged**